src/Features/TestUtilities/EditAndContinue/EditAndContinueTestHelpers.cs

```diff
@@ -506,6 +506,17 @@ public static MatchingPairs ToMatchingPairs(IEnumerable<KeyValuePair<SyntaxNode,
                     New = partners.Value.ToString().Replace("\r\n", " ").Replace("\n", " ")
                 }));
         }
+
+        public static void SetDocumentsState(DebuggingSession session, Solution solution, CommittedSolution.DocumentState state)
+        {
+            foreach (var project in solution.Projects)
+            {
+                foreach (var document in project.Documents)
+                {
+                    session.LastCommittedSolution.Test_SetDocumentState(document.Id, state);
```

Comment:
As a follow-up, this should go through a TestAccessor (which has analyzer validation that it doesn't get called from production code)